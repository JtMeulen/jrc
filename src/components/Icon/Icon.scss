@use '../../style/lib.scss' as *;
@use '../../style/mixins.scss';

.icon {
  background-color: $jrc-color-font;

  &--size-sm {
    width: 24px;
    height: 24px;
  }
  &--size-md {
    width: 36px;
    height: 36px;
  }
  &--size-lg {
    width: 48px;
    height: 48px;
  }

  &--check {
    mask-image: url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='utf-8'%3F%3E%3C!-- Generator: Adobe Illustrator 23.0.1  SVG Export Plug-In . SVG Version: 6.00 Build 0) --%3E%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' x='0px' y='0px' viewBox='0 0 512 512' style='enable-background:new 0 0 512 512%3B' xml:space='preserve'%3E%3Cstyle type='text/css'%3E.st0%7Bfill: %7D%3C/style%3E%3Cg id='Layer_1'%3E%3C/g%3E%3Cg id='Layer_2'%3E%3Cg%3E%3Cpath class='st0' d='M256 27.5c-61.03 0-118.42 23.77-161.57 66.93C51.27 137.58 27.5 194.97 27.5 256s23.77 118.42 66.93 161.57C137.58 460.73 194.97 484.5 256 484.5s118.42-23.77 161.57-66.93C460.73 374.42 484.5 317.03 484.5 256s-23.77-118.42-66.93-161.57C374.42 51.27 317.03 27.5 256 27.5z M256 452.5c-108.35 0-196.5-88.15-196.5-196.5S147.65 59.5 256 59.5S452.5 147.65 452.5 256S364.35 452.5 256 452.5z'/%3E%3Cpath class='st0' d='M347.7 131.81c-16.03 0-31.09 6.24-42.43 17.57l-85.16 85.16l-13.39-13.39c-11.33-11.33-26.4-17.58-42.43-17.58c-16.03 0-31.1 6.24-42.43 17.57c-11.32 11.32-17.56 26.39-17.56 42.43s6.24 31.1 17.56 42.43l57.17 57.17c10.97 10.97 25.56 17.01 41.08 17.01s30.1-6.04 41.08-17.01l128.94-128.94c11.32-11.32 17.56-26.39 17.56-42.43c0-16.04-6.24-31.1-17.56-42.42C378.8 138.05 363.73 131.81 347.7 131.81z M367.5 211.61L238.57 340.55c-4.93 4.93-11.48 7.64-18.45 7.64s-13.52-2.71-18.45-7.64l-57.17-57.17c-5.28-5.28-8.18-12.31-8.18-19.8s2.91-14.52 8.19-19.8c5.29-5.29 12.32-8.2 19.8-8.2s14.51 2.91 19.8 8.2l24.71 24.7c6.25 6.25 16.38 6.25 22.63 0l96.47-96.47c5.29-5.29 12.32-8.2 19.8-8.2s14.51 2.91 19.8 8.2c5.28 5.28 8.18 12.31 8.18 19.8S372.78 206.33 367.5 211.61z'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E");
    -webkit-mask-image: url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='utf-8'%3F%3E%3C!-- Generator: Adobe Illustrator 23.0.1  SVG Export Plug-In . SVG Version: 6.00 Build 0) --%3E%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' x='0px' y='0px' viewBox='0 0 512 512' style='enable-background:new 0 0 512 512%3B' xml:space='preserve'%3E%3Cstyle type='text/css'%3E.st0%7Bfill: %7D%3C/style%3E%3Cg id='Layer_1'%3E%3C/g%3E%3Cg id='Layer_2'%3E%3Cg%3E%3Cpath class='st0' d='M256 27.5c-61.03 0-118.42 23.77-161.57 66.93C51.27 137.58 27.5 194.97 27.5 256s23.77 118.42 66.93 161.57C137.58 460.73 194.97 484.5 256 484.5s118.42-23.77 161.57-66.93C460.73 374.42 484.5 317.03 484.5 256s-23.77-118.42-66.93-161.57C374.42 51.27 317.03 27.5 256 27.5z M256 452.5c-108.35 0-196.5-88.15-196.5-196.5S147.65 59.5 256 59.5S452.5 147.65 452.5 256S364.35 452.5 256 452.5z'/%3E%3Cpath class='st0' d='M347.7 131.81c-16.03 0-31.09 6.24-42.43 17.57l-85.16 85.16l-13.39-13.39c-11.33-11.33-26.4-17.58-42.43-17.58c-16.03 0-31.1 6.24-42.43 17.57c-11.32 11.32-17.56 26.39-17.56 42.43s6.24 31.1 17.56 42.43l57.17 57.17c10.97 10.97 25.56 17.01 41.08 17.01s30.1-6.04 41.08-17.01l128.94-128.94c11.32-11.32 17.56-26.39 17.56-42.43c0-16.04-6.24-31.1-17.56-42.42C378.8 138.05 363.73 131.81 347.7 131.81z M367.5 211.61L238.57 340.55c-4.93 4.93-11.48 7.64-18.45 7.64s-13.52-2.71-18.45-7.64l-57.17-57.17c-5.28-5.28-8.18-12.31-8.18-19.8s2.91-14.52 8.19-19.8c5.29-5.29 12.32-8.2 19.8-8.2s14.51 2.91 19.8 8.2l24.71 24.7c6.25 6.25 16.38 6.25 22.63 0l96.47-96.47c5.29-5.29 12.32-8.2 19.8-8.2s14.51 2.91 19.8 8.2c5.28 5.28 8.18 12.31 8.18 19.8S372.78 206.33 367.5 211.61z'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E");
  }

  &--star {
    mask-image: url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='utf-8'%3F%3E%3C!-- Generator: Adobe Illustrator 23.0.1  SVG Export Plug-In . SVG Version: 6.00 Build 0) --%3E%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' x='0px' y='0px' viewBox='0 0 512 512' style='enable-background:new 0 0 512 512%3B' xml:space='preserve'%3E%3Cstyle type='text/css'%3E.st0%7Bfill:%23333333%3B%7D%3C/style%3E%3Cg id='Layer_1'%3E%3C/g%3E%3Cg id='Layer_2'%3E%3Cg%3E%3Cpath class='st0' d='M451.29 176.95l-112.05-10.13c-1.63-0.15-3.05-1.18-3.7-2.69L291.29 60.7c0 0 0 0 0 0c-5.84-13.64-18.68-22.12-33.52-22.12s-27.68 8.48-33.52 22.12L180 164.14c-0.65 1.51-2.06 2.54-3.7 2.69L64.25 176.95c-14.78 1.34-26.81 10.93-31.39 25.04c-4.58 14.11-0.49 28.95 10.68 38.71l84.7 74.05c1.24 1.08 1.78 2.75 1.41 4.35l-25 109.7c-3.3 14.47 2.11 28.87 14.12 37.59c6.46 4.7 13.9 7.06 21.38 7.06c6.41 0 12.86-1.74 18.74-5.25l96.6-57.67c1.41-0.84 3.16-0.84 4.57 0l96.6 57.67c12.74 7.61 28.11 6.91 40.12-1.81c12-8.72 17.42-23.13 14.12-37.6l-25-109.7c-0.37-1.6 0.18-3.27 1.41-4.35l84.7-74.05c11.17-9.77 15.26-24.6 10.68-38.71C478.1 187.88 466.07 178.28 451.29 176.95z M450.95 216.61l-84.7 74.05c-10.11 8.84-14.53 22.46-11.55 35.55l25 109.7c0.54 2.37-0.64 3.81-1.73 4.6c-1.08 0.79-2.82 1.47-4.91 0.22l-96.6-57.67c-11.53-6.88-25.85-6.88-37.38 0l-96.6 57.67c-2.09 1.24-3.82 0.57-4.91-0.22c-1.09-0.79-2.27-2.23-1.73-4.6l25-109.7c2.98-13.09-1.44-26.71-11.55-35.55l-84.7-74.05c-1.83-1.6-1.72-3.46-1.31-4.73c0.42-1.28 1.42-2.84 3.84-3.06l112.05-10.13c13.37-1.21 24.96-9.63 30.24-21.97l44.26-103.44c0.96-2.23 2.76-2.7 4.1-2.7s3.14 0.47 4.1 2.7l44.26 103.44c5.28 12.34 16.87 20.76 30.24 21.97l112.05 10.13c2.42 0.22 3.42 1.79 3.84 3.06C452.67 213.16 452.78 215.01 450.95 216.61z'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E");
    -webkit-mask-image: url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='utf-8'%3F%3E%3C!-- Generator: Adobe Illustrator 23.0.1  SVG Export Plug-In . SVG Version: 6.00 Build 0) --%3E%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' x='0px' y='0px' viewBox='0 0 512 512' style='enable-background:new 0 0 512 512%3B' xml:space='preserve'%3E%3Cstyle type='text/css'%3E.st0%7Bfill:%23333333%3B%7D%3C/style%3E%3Cg id='Layer_1'%3E%3C/g%3E%3Cg id='Layer_2'%3E%3Cg%3E%3Cpath class='st0' d='M451.29 176.95l-112.05-10.13c-1.63-0.15-3.05-1.18-3.7-2.69L291.29 60.7c0 0 0 0 0 0c-5.84-13.64-18.68-22.12-33.52-22.12s-27.68 8.48-33.52 22.12L180 164.14c-0.65 1.51-2.06 2.54-3.7 2.69L64.25 176.95c-14.78 1.34-26.81 10.93-31.39 25.04c-4.58 14.11-0.49 28.95 10.68 38.71l84.7 74.05c1.24 1.08 1.78 2.75 1.41 4.35l-25 109.7c-3.3 14.47 2.11 28.87 14.12 37.59c6.46 4.7 13.9 7.06 21.38 7.06c6.41 0 12.86-1.74 18.74-5.25l96.6-57.67c1.41-0.84 3.16-0.84 4.57 0l96.6 57.67c12.74 7.61 28.11 6.91 40.12-1.81c12-8.72 17.42-23.13 14.12-37.6l-25-109.7c-0.37-1.6 0.18-3.27 1.41-4.35l84.7-74.05c11.17-9.77 15.26-24.6 10.68-38.71C478.1 187.88 466.07 178.28 451.29 176.95z M450.95 216.61l-84.7 74.05c-10.11 8.84-14.53 22.46-11.55 35.55l25 109.7c0.54 2.37-0.64 3.81-1.73 4.6c-1.08 0.79-2.82 1.47-4.91 0.22l-96.6-57.67c-11.53-6.88-25.85-6.88-37.38 0l-96.6 57.67c-2.09 1.24-3.82 0.57-4.91-0.22c-1.09-0.79-2.27-2.23-1.73-4.6l25-109.7c2.98-13.09-1.44-26.71-11.55-35.55l-84.7-74.05c-1.83-1.6-1.72-3.46-1.31-4.73c0.42-1.28 1.42-2.84 3.84-3.06l112.05-10.13c13.37-1.21 24.96-9.63 30.24-21.97l44.26-103.44c0.96-2.23 2.76-2.7 4.1-2.7s3.14 0.47 4.1 2.7l44.26 103.44c5.28 12.34 16.87 20.76 30.24 21.97l112.05 10.13c2.42 0.22 3.42 1.79 3.84 3.06C452.67 213.16 452.78 215.01 450.95 216.61z'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E");
  }

  &--chevron-right {
    mask-image: url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='utf-8'%3F%3E%3C!-- Generator: Adobe Illustrator 23.0.1  SVG Export Plug-In . SVG Version: 6.00 Build 0) --%3E%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' x='0px' y='0px' viewBox='0 0 512 512' style='enable-background:new 0 0 512 512%3B' xml:space='preserve'%3E%3Cstyle type='text/css'%3E.st0%7Bfill: %3B%7D%3C/style%3E%3Cg id='Layer_1'%3E%3C/g%3E%3Cg id='Layer_2'%3E%3Cg%3E%3Cpath class='st0' d='M256 27.5c-61.03 0-118.42 23.77-161.57 66.93C51.27 137.58 27.5 194.97 27.5 256s23.77 118.42 66.93 161.57C137.58 460.73 194.97 484.5 256 484.5s118.42-23.77 161.57-66.93C460.73 374.42 484.5 317.03 484.5 256s-23.77-118.42-66.93-161.57C374.42 51.27 317.03 27.5 256 27.5z M256 452.5c-108.35 0-196.5-88.15-196.5-196.5S147.65 59.5 256 59.5S452.5 147.65 452.5 256S364.35 452.5 256 452.5z'/%3E%3Cpath class='st0' d='M210.47 131.16c-6.25-6.25-16.38-6.25-22.63 0c-6.25 6.25-6.25 16.38 0 22.63L290.2 256.15L187.86 358.2c-6.26 6.24-6.27 16.37-0.03 22.63c3.13 3.13 7.23 4.7 11.33 4.7c4.09 0 8.17-1.56 11.3-4.67L324.14 267.5c3.01-3 4.7-7.07 4.7-11.32s-1.68-8.32-4.69-11.33L210.47 131.16z'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E");
    -webkit-mask-image: url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='utf-8'%3F%3E%3C!-- Generator: Adobe Illustrator 23.0.1  SVG Export Plug-In . SVG Version: 6.00 Build 0) --%3E%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' x='0px' y='0px' viewBox='0 0 512 512' style='enable-background:new 0 0 512 512%3B' xml:space='preserve'%3E%3Cstyle type='text/css'%3E.st0%7Bfill: %3B%7D%3C/style%3E%3Cg id='Layer_1'%3E%3C/g%3E%3Cg id='Layer_2'%3E%3Cg%3E%3Cpath class='st0' d='M256 27.5c-61.03 0-118.42 23.77-161.57 66.93C51.27 137.58 27.5 194.97 27.5 256s23.77 118.42 66.93 161.57C137.58 460.73 194.97 484.5 256 484.5s118.42-23.77 161.57-66.93C460.73 374.42 484.5 317.03 484.5 256s-23.77-118.42-66.93-161.57C374.42 51.27 317.03 27.5 256 27.5z M256 452.5c-108.35 0-196.5-88.15-196.5-196.5S147.65 59.5 256 59.5S452.5 147.65 452.5 256S364.35 452.5 256 452.5z'/%3E%3Cpath class='st0' d='M210.47 131.16c-6.25-6.25-16.38-6.25-22.63 0c-6.25 6.25-6.25 16.38 0 22.63L290.2 256.15L187.86 358.2c-6.26 6.24-6.27 16.37-0.03 22.63c3.13 3.13 7.23 4.7 11.33 4.7c4.09 0 8.17-1.56 11.3-4.67L324.14 267.5c3.01-3 4.7-7.07 4.7-11.32s-1.68-8.32-4.69-11.33L210.47 131.16z'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E");
  }

  &--chevron-left {
    mask-image: url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='utf-8'%3F%3E%3C!-- Generator: Adobe Illustrator 23.0.1  SVG Export Plug-In . SVG Version: 6.00 Build 0) --%3E%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' x='0px' y='0px' viewBox='0 0 512 512' style='enable-background:new 0 0 512 512%3B' xml:space='preserve'%3E%3Cstyle type='text/css'%3E.st0%7Bfill: %3B%7D%3C/style%3E%3Cg id='Layer_1'%3E%3C/g%3E%3Cg id='Layer_2'%3E%3Cg%3E%3Cpath class='st0' d='M256 27.5c-61.03 0-118.42 23.77-161.57 66.93C51.27 137.58 27.5 194.97 27.5 256s23.77 118.42 66.93 161.57C137.58 460.73 194.97 484.5 256 484.5s118.42-23.77 161.57-66.93C460.73 374.42 484.5 317.03 484.5 256s-23.77-118.42-66.93-161.57C374.42 51.27 317.03 27.5 256 27.5z M256 452.5c-108.35 0-196.5-88.15-196.5-196.5S147.65 59.5 256 59.5S452.5 147.65 452.5 256S364.35 452.5 256 452.5z'/%3E%3Cpath class='st0' d='M324.16 131.16c-6.25-6.25-16.38-6.25-22.63 0L187.84 244.85c-3 3-4.69 7.08-4.69 11.33s1.7 8.32 4.7 11.32l113.69 113.36c3.12 3.11 7.21 4.67 11.3 4.67c4.1 0 8.21-1.57 11.33-4.7c6.24-6.26 6.22-16.39-0.03-22.63L221.8 256.15l102.36-102.36C330.41 147.54 330.41 137.41 324.16 131.16z'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E");
    -webkit-mask-image: url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='utf-8'%3F%3E%3C!-- Generator: Adobe Illustrator 23.0.1  SVG Export Plug-In . SVG Version: 6.00 Build 0) --%3E%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' x='0px' y='0px' viewBox='0 0 512 512' style='enable-background:new 0 0 512 512%3B' xml:space='preserve'%3E%3Cstyle type='text/css'%3E.st0%7Bfill: %3B%7D%3C/style%3E%3Cg id='Layer_1'%3E%3C/g%3E%3Cg id='Layer_2'%3E%3Cg%3E%3Cpath class='st0' d='M256 27.5c-61.03 0-118.42 23.77-161.57 66.93C51.27 137.58 27.5 194.97 27.5 256s23.77 118.42 66.93 161.57C137.58 460.73 194.97 484.5 256 484.5s118.42-23.77 161.57-66.93C460.73 374.42 484.5 317.03 484.5 256s-23.77-118.42-66.93-161.57C374.42 51.27 317.03 27.5 256 27.5z M256 452.5c-108.35 0-196.5-88.15-196.5-196.5S147.65 59.5 256 59.5S452.5 147.65 452.5 256S364.35 452.5 256 452.5z'/%3E%3Cpath class='st0' d='M324.16 131.16c-6.25-6.25-16.38-6.25-22.63 0L187.84 244.85c-3 3-4.69 7.08-4.69 11.33s1.7 8.32 4.7 11.32l113.69 113.36c3.12 3.11 7.21 4.67 11.3 4.67c4.1 0 8.21-1.57 11.33-4.7c6.24-6.26 6.22-16.39-0.03-22.63L221.8 256.15l102.36-102.36C330.41 147.54 330.41 137.41 324.16 131.16z'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E");
  }
}
