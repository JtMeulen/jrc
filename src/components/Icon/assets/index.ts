export { default as check } from "./check.svg";
