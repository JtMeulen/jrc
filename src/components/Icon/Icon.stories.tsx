import React from "react";
import { Meta, StoryFn } from "@storybook/react";
import Icon from "./Icon";

export default {
  title: "Components/Icon",
  component: Icon,
  argTypes: {
    name: {
      control: "select",
      options: ["chevron-left", "chevron-right", "error", "info", "star", "success", "warning"],
    },
  },
  args: { name: 'star' },
} as Meta<typeof Icon>;

const Template: StoryFn<typeof Icon> = (args) => <Icon {...args} />;

export const Default = Template.bind({});
Default.args = {};
