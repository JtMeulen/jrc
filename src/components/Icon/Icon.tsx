import React from "react";
import cn from "classnames";

import "./Icon.scss";

export interface IconProps {
  name: string;
  size?: "sm" | "md" | "lg";
  color?: string;
  title?: string;
}

const Icon = ({ name, size = "md", color, title }: IconProps) => {
  return (
    <div
      className={cn("icon", `icon--size-${size}`, `icon--${name}`)}
      style={{ backgroundColor: color }}
      aria-label={title}
    ></div>
  );
};

export default Icon;
