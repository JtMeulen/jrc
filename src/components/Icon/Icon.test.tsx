import React from "react";
import { render, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Icon from "./Icon";

describe("Icon", () => {
  test("renders without error", () => {
    const { container } = render(<Icon name="success" />);
    expect(container).toMatchSnapshot();
  });

  test("renders with different props", async () => {
    const { container } = render(
      <Icon name="star" size="lg" title="aria text" color="red" />
    );
    expect(container).toMatchSnapshot();
  });
});
