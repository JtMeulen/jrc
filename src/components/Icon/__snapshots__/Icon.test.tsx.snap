// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders with different props 1`] = `
<div>
  <div
    aria-label="aria text"
    class="icon icon--size-lg icon--star"
    style="background-color: red;"
  />
</div>
`;

exports[`Icon renders without error 1`] = `
<div>
  <div
    class="icon icon--size-md icon--success"
  />
</div>
`;
